src/library/scala/collection/Seq.scala

```diff
@@ -920,8 +929,11 @@ trait SeqOps[+A, +CC[_], +C] extends Any
   }
 
   protected[collection] def occCounts[B](sq: Seq[B]): mutable.Map[B, Int] = {
-    val occ = new mutable.HashMap[B, Int]().withDefaultValue(0)
-    for (y <- sq) occ(y) += 1
+    val occ = new mutable.HashMap[B, Int]()
+    for (y <- sq) occ.updateWith(y) {
+      case None => Some(1)
+      case Some(n) => Some(n + 1)
+    }
```

Comment:
Maybe we could avoid some allocations if we cache the Some's up to (for example..) 10? 

```scala
private object CachedSome {
  private[this] val cached: Array[Some[Int]] = Array.tabulate(10)(Some(_))
  def apply(i: Int): if (0 <= i && i < 10) cached(i) else Some(i)
}
```

Reply:
I don't trust my laptop with the benchmarks, or, honestly, want to dedicate it to running them (I don't think that https://github.com/scala/scala-dev/issues/753 was opened shortly after @NthPortal  looked at this PR is a coincidence).

If this is worth it, it's plausible it's worth it globally, on Some.apply, so that any map that keeps a count of things takes advantage of it (though at the cost of a probably correctly branch-predicted instanceOf check), which would be orthogonal of this PR.

On the gripping hand, maybe that's me internally rationalizing my way out of just writing and running the damn benchmarks.